### PR TITLE
Help Center: Add site type check on help center

### LINKED ISSUE #65974: Avoid calling useHasSeenWhatsNewModalQuery for atomic
We use this query [here](https://github.com/Automattic/wp-calypso/blob/6264558dc849d9e9f0d8a776e1d7aca2ab2fae4a/apps/editing-toolkit/editing-toolkit-plugin/help-center/src/help-center.js#L26) and [here](https://github.com/Automattic/wp-calypso/blob/6264558dc849d9e9f0d8a776e1d7aca2ab2fae4a/packages/help-center/src/components/help-center-more-resources.tsx#L44)

The request fails with 404 error
Screenshot 2022-07-26 at 17 26 24

As a temporary fix we should disable this call for atomic sites.

### PR DESCRIPTION
**Pull request by renancarvalho**
#### Proposed Changes
Add a site type check on help-center.
If site is atomic it should never perform a request to `https://public-api.wordpress.com/wpcom/v2/sites/208767849/block-editor/has-seen-whats-new-modal?_envelope=1` endpoint.

No requests on atomic sites
image

Requests on simple sites
image

#### Testing Instructions
* Pull this branch `fix/add-site-type-check-help-center`
* Run `yarn start`
* Navigate to your site, for instance `http://calypso.localhost:3000/post/yoursite.com`
* Open the network tab and check for api requests to `wpcom/v2/sites/208767849/block-editor/has-seen-whats-new-modal`

Check: If your site is a simple site, you should see the requests. If atomic you should not

Related to #65974
Fixes #65974

### CODE REVIEW

File: apps/editing-toolkit/editing-toolkit-plugin/help-center/src/help-center.js [hunk 1]
```diff
@@ -23,7 +24,10 @@ function HelpCenterContent() {
 	const [ showHelpIcon, setShowHelpIcon ] = useState( false );
 	const { setShowHelpCenter } = useDispatch( 'automattic/help-center' );
 	const [ showHelpIconDot, setShowHelpIconDot ] = useState( false );
-	const { data, isLoading } = useHasSeenWhatsNewModalQuery( window._currentSiteId );
+	const isSimpleSite = useSelector( ( state ) => getIsSimpleSite( state ) ) || false;
```
Comment (escapemanuele) [inline apps/editing-toolkit/editing-toolkit-plugin/help-center/src/help-center.js]: You can write this as `const isSimpleSite = useSelector( getIsSimpleSite ) || false;`

File: client/layout/masterbar/checkout.tsx [hunk 2]
```diff
@@ -40,8 +41,8 @@ const CheckoutMasterbar = ( {
 	const translate = useTranslate();
 	const jetpackCheckoutBackUrl = useValidCheckoutBackUrl( siteSlug );
 	const siteId = useSelector( getSelectedSiteId );
-
-	const { isLoading, data } = useHasSeenWhatsNewModalQuery( siteId );
+	const isSimpleSite = useSelector( ( state ) => getIsSimpleSite( state ) ) || false;
```
Comment (escapemanuele) [inline client/layout/masterbar/checkout.tsx]: Same thing here!

File: packages/data-stores/src/queries/use-has-seen-whats-new-modal-query.ts [hunk 3]
```diff
@@ -26,7 +26,7 @@ export const useHasSeenWhatsNewModalQuery = ( siteId: number | null ) => {
 				apiNamespace: 'wpcom/v2',
 			} ),
 		{
-			enabled: !! siteId,
+			enabled: isSimpleSite,
```
Comment (escapemanuele) [inline packages/data-stores/src/queries/use-has-seen-whats-new-modal-query.ts]: Why did you remove the `!! siteId` check? Could this be  `!! siteId && isSimpleSite`?

File: packages/help-center/src/components/help-center-more-resources.tsx [hunk 4]
```diff
@@ -26,22 +26,26 @@ export const HelpCenterMoreResources = () => {
...
 				purchaseSlugs &&
 				( purchaseSlugs.some( isWpComBusinessPlan ) || purchaseSlugs.some( isWpComEcommercePlan ) )
 			),
-			isSimpleSite: site && ! site.is_wpcom_atomic,
 			siteId: siteId,
 		};
 	} );
-	const { data, isLoading, setHasSeenWhatsNewModal } = useHasSeenWhatsNewModalQuery( siteId );
+
+	const isSimpleSite = useSelector( ( state ) => getIsSimpleSite( state ) ) || false;
```
Comment (escapemanuele) [inline packages/help-center/src/components/help-center-more-resources.tsx]: Same as before! Moreover, this could be moved inside the other `useSelector`!

**Review by escapemanuele — APPROVED** [review]
This works, thanks a lot @renancarvalho, great job 💯 
I left some suggestions that I'd love to be addressed before accepting this!

File: packages/help-center/src/components/help-center-more-resources.tsx (hunk 4 above)
Comment (renancarvalho) [reply packages/help-center/src/components/help-center-more-resources.tsx]: Thank you for reviewing @escapemanuele , I have applied your good suggestions. 🙇‍♂️

**Comment by escapemanuele** [comment]
Wonderful, thanks for addressing my feedback! Deploy this 🥇
